src/nix-env/nix-env.cc

```diff
@@ -564,17 +577,23 @@ static void installDerivations(Globals & globals,
 
 static void opInstall(Globals & globals, Strings opFlags, Strings opArgs)
 {
+    std::optional<int> priority;
     for (Strings::iterator i = opFlags.begin(); i != opFlags.end(); ) {
         auto arg = *i++;
         if (parseInstallSourceOptions(globals, i, opFlags, arg)) ;
         else if (arg == "--preserve-installed" || arg == "-P")
             globals.preserveInstalled = true;
         else if (arg == "--remove-all" || arg == "-r")
             globals.removeAll = true;
+        else if (arg == "--priority") {
+            if (i == opFlags.end())
+                throw UsageError("'%1%' requires an argument", arg);
+            priority = std::stoi(*i++);
```

Comment:
You may want to use `string2Int` to get a better error message than `std::invalid_argument`.
```suggestion
            priority = string2Int<int>(*i++);
            if (!priority)
                throw UsageError("'--priority' requires an integer argument");
```

Reply:
Good call